### PR TITLE
Add status bar item

### PR DESCRIPTION
**Pull request by mburleigh**
Add spinner to status bar while the AZ command is executing. Show command duration when execution is finished and hide 10 seconds after command is finished executing.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -147,22 +145,37 @@ class RunLineInTerminal {
     }
 }
 
+const elegantSpinner = require('elegant-spinner');
```
Comment (chrmarti) [inline src/extension.ts]: Move to import section at the top of the file and use `import` instead of `require`.

File: src/extension.ts [hunk 2]
```diff
@@ -147,22 +145,37 @@ class RunLineInTerminal {
...
         this.disposables.push(commands.registerTextEditorCommand('ms-azurecli.runLineInEditor', editor => this.run(editor)));
         this.disposables.push(workspace.onDidCloseTextDocument(document => this.close(document)));
         this.disposables.push(workspace.onDidChangeTextDocument(event => this.change(event)));
+
+        this.runStatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left);
     }
 
     private run(source: TextEditor) {
+        const t0 = Date.now();
+        this.interval = setInterval(() => {
```
Comment (chrmarti) [inline src/extension.ts]: This does not yet handle the case where the user executes a second command before the first has finished.

File: src/extension.ts [hunk 3]
```diff
@@ -172,14 +185,24 @@ class RunLineInEditor {
             .then(target => replaceContent(target, JSON.stringify({ 'Running command': line }) + '\n')
                 .then(() => exec(line))
                 .then(({ stdout }) => stdout, ({ stdout, stderr }) => JSON.stringify({ stderr, stdout }, null, '    '))
-                .then(content => replaceContent(target, content)
-                    .then(() => this.parsedResult = JSON.parse(content))
-                    .then(undefined, err => {})
-                )
+                .then(content => {
+                    replaceContent(target, content)
```
Comment (chrmarti) [inline src/extension.ts]: `return` to keep the promise chaining? Or move the two lines below to `hideStatusBarItem()` and keep this unchanged.

File: src/extension.ts [hunk 4]
```diff
@@ -242,6 +265,7 @@ class RunLineInEditor {
 
     dispose() {
         this.disposables.forEach(disposable => disposable.dispose());
+        this.runStatusBarItem.dispose();
```
Comment (chrmarti) [inline src/extension.ts]: Instead add the status bar item to `this.disposables`.

File: src/extension.ts (hunk 1 above)
Comment (mburleigh) [reply src/extension.ts]: Looks like this will require me to create a new .d.ts file. Do you have any preference where that file should go?

Comment (chrmarti) [reply src/extension.ts]: The latest version of elegant-spinner comes with a .d.ts. Have you tried that?

Comment (mburleigh) [reply src/extension.ts]: I did:
`C:\Users\mburleigh\Source\Repos\vscode-azurecli-mfb>npm install @types/elegant-spinner
npm WARN deprecated @types/elegant-spinner@2.0.1: This is a stub types definition. elegant-spinner provides its own type definitions, so you do not need this installed.
+ @types/elegant-spinner@2.0.1
added 1 package in 2.515s`
after that looking in the \node_modules\@types\elegant-spinner directory there is no index.d.ts file.

Comment (chrmarti) [reply src/extension.ts]: You don't need the @types package, but the 2.0 release of the main package from what I see.

Comment (mburleigh) [reply src/extension.ts]: I'll take a look; thanks for the feedback!

File: src/extension.ts (hunk 4 above)
Comment (mburleigh) [reply src/extension.ts]: Fixed

File: src/extension.ts (hunk 3 above)
Comment (mburleigh) [reply src/extension.ts]: Fixed

File: src/extension.ts (hunk 1 above)
Comment (mburleigh) [reply src/extension.ts]: Fixed. Had to uninstall, clean up my node_modules, & reinstall to get the new version.

File: src/extension.ts (hunk 2 above)
Comment (mburleigh) [reply src/extension.ts]: Added some code to support commands running simultaneously. Keeping count of how many commands are running (runningCommandCount) feels a little kludgy to me but I all the alternatives I tried were worse.

**Comment by chrmarti** [comment]
Nice! I'm tweaking the messages a little so they read 'Azure CLI' (I don't think we 'AZ CLI' elsewhere) and are slightly shorter. With that I'll merge it for the release I'm about to make. Hope that's fine with you.

Thanks for the PR!

**Pull request merged**